draconic/helpers.py

```diff
@@ -155,6 +161,19 @@ def _safe_sub(self, a, b):
             _raise_in_context(NumberTooHigh, "Subtracting these two would create a number too large")
         return result
 
+    def _safe_lshift(self, a, b):
+        """Left Bit-Shift: limit the size of integers/floats to prevent CPU-locking computation"""
+        self._check_binop_operands(a, b)
+
+        if abs(a) > self._config.max_power_base or abs(b) > self._config.max_power:
```

Comment:
Since `a << b` is equivalent to `a * (2 ** b)`, what we're actually checking is that `2 ** b` is within the multiplicative limits (`a` is already safe from the `_check_binop_operands` call).

Copying from `safe_mult`, that check is `if isinstance(b, int) and (2 ** b < self._config.min_int or 2 ** b > self._config.max_int)`, but we can simplify this knowing:
- `b` cannot be negative (since lshift only works on positive integers)
- `max_int` is defined as `2 ** (max_int_size - 1) - 1`

So really, the check here is `if isinstance(b, int) and b > self._config.max_int_size - 2`.

Reply:
lshift actually does work on negative integers

```py
-1 << 1
>>> -2
-1 <<4
>>> -16
```

Reply:
Wait I misunderstood, ignore the above

Reply:
A more useful question, it seems that `max_int_size` isn't exposed in DraconicConfig,
Should I add that into the PR?

Reply:
It looks like this PR is based on a slightly outdated version of `master`. Try updating from that, and then yes.

Reply:
I might be going insane, but it looks like `master` doesn't have it either.